galvasr2/ner_pipeline_transcript.py

```diff
@@ -0,0 +1,128 @@
+import pandas as pd
+from flair.data import Sentence
+from flair.models import SequenceTagger
+import logging
+from absl import flags
+from multiprocessing.pool import ThreadPool
+from pyspark.sql import SparkSession
+from tqdm import tqdm
+
+from galvasr2.align.spark.align_lib import (
+    load_audio_id_text_id_mapping,
+    load_transcripts,
+)
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    "input_transcription_index",
+    "gs://the-peoples-speech-west-europe/archive_org/Mar_7_2021/CC_BY_SA_EXPANDED_LICENSES_FILTERED_ACCESS.jsonl.gz",
+    "Ubication of the path with the transcription index data",
+)
+flags.DEFINE_string(
+    "input_transcription_data",
+    "gs://the-peoples-speech-west-europe/archive_org/Mar_7_2021/CC_BY_SA_EXPANDED_LICENSES_FILTERED_ACCESS",
+    "Ubication of the path with the transcription data",
+)
+flags.DEFINE_string(
+    "num_rows",
+    10,
+    "Number of transcripts to analyze",
+)
+
+
+class NER:
+    def __init__(self, num_rows: int):
+        self.num_rows = num_rows
+
+    def read_transcription_data(
+        self,
+        spark: SparkSession,
+        data_trans_index: str,
+        data_trans: str,
+    ) -> pd.DataFrame:
+        # spark.sparkContext.setLogLevel("INFO") # "ALL" for very verbose logging
+        logging.getLogger("py4j").setLevel(logging.ERROR)
+        catalogue_df = load_audio_id_text_id_mapping(spark, data_trans_index)
+        training_sample_rows = catalogue_df.collect()
+
+        # Comment this out to load everything. It might takes ~15 minute, in my experience, on an 8 core machine.
+        training_sample_rows = training_sample_rows[: self.num_rows]
+        transcripts_df = load_transcripts(spark, data_trans, training_sample_rows)
+        transcripts_pdf = transcripts_df.toPandas()
+        return transcripts_pdf
+
+    def load_ner_model(self) -> SequenceTagger:
+        model = SequenceTagger.load("flair/ner-english-ontonotes-fast")
+        return model
+
+    def get_ner_transcription(self, transcripts_pdf: pd.DataFrame) -> dict:
+        ner_entities = {
```

Comment:
As an FYI, if your set of keys isn't known in advanced collections.defaultdict(list) is a powerful tool you can use. I don't recommend changing this code, though. It is helpful to have the tags visible for understanding.

Reply:
Good suggestion, thanks